partner_firstname/tests/test_onchange.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""These tests try to mimic the behavior of the UI form.
+
+The form operates in onchange mode, and has only some limitations that must be
+met.
+"""
+
+from openerp.tests.common import TransactionCase
+
+
+class PartnerCompanyCase(TransactionCase):
+    def test_create_from_form(self):
+        """A user creates a company from the form."""
+        name = u"Sôme company"
+        with self.env.do_in_onchange():
+            # User presses ``new``
+            partner = self.env["res.partner"].create({})
```

Comment:
This (and others equal) is the failing line, but I cannot reproduce the error in my tests, with an empty database. When a user presses "New", an empty record is created in an onchange environment, and that's exactly what I do here. Sholud I call a different method?